amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/AddToHoldsBulkV1Tests.java

```diff
@@ -0,0 +1,519 @@
+import static org.junit.Assert.assertTrue;
+import static org.springframework.http.HttpStatus.ACCEPTED;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import org.alfresco.dataprep.CMISUtil;
+import org.alfresco.dataprep.ContentActions;
+import org.alfresco.rest.rm.community.base.BaseRMRestTest;
+import org.alfresco.rest.rm.community.model.hold.Hold;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkOperation;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkOperation.HoldBulkOperationType;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkOperationEntry;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatus;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatus.Status;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatusCollection;
+import org.alfresco.rest.rm.community.model.hold.HoldChild;
+import org.alfresco.rest.rm.community.model.hold.HoldChildEntry;
+import org.alfresco.rest.rm.community.model.user.UserRoles;
+import org.alfresco.rest.search.RestRequestQueryModel;
+import org.alfresco.rest.search.SearchRequest;
+import org.alfresco.rest.v0.service.RoleService;
+import org.alfresco.utility.constants.UserRole;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * API tests for adding items to holds via the bulk process
+ */
+public class AddToHoldsBulkV1Tests extends BaseRMRestTest
+{
+    private static final String ACCESS_DENIED_ERROR_MESSAGE = "Access Denied.  You do not have the appropriate " +
+        "permissions to perform this operation.";
+    private static final int NUMBER_OF_FILES = 30;
+    private final List<FileModel> addedFiles = new ArrayList<>();
+    private final List<UserModel> users = new ArrayList<>();
+    private final List<Hold> holds = new ArrayList<>();
+    private Hold hold;
+    private Hold hold2;
+    private Hold hold3;
+    private FolderModel rootFolder;
+    private HoldBulkOperation holdBulkOperation;
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private ContentActions contentActions;
+
+    @BeforeClass(alwaysRun = true)
+    public void preconditionForAddContentToHold()
+    {
+        STEP("Create a hold.");
+        hold = getRestAPIFactory().getFilePlansAPI(getAdminUser()).createHold(
+            Hold.builder().name("HOLD" + generateTestPrefix(AddToHoldsV1Tests.class)).description(HOLD_DESCRIPTION)
```

Comment:
In theory the idea was to create a constant:
```
private static final String PREFIX = generateTestPrefix(AddToHoldsV1Tests.class);
```
and then use it as a prefix:
```
.name(PREFIX + "HOLD")
```
but it doesn't really matter if you just want a random name.

Reply:
I left it as it is because in this test class we need 3 unique hold names.